Add support for 24hr-averaged ocn/ice products

EricSinsky-NOAA opened the pull request:
<!--
  *** PLEASE READ ***

  Any PRs not following this template will be closed.

  Please delete all these comments before submitting the PR.

  Please use a short (<60 char), descriptive title for the PR title above. It should complete the sentence "If merged, this PR will _____". Capitalize the first word and do not end with a period.

  No content should appear above the "Description" header.

  If this PR is not merge-ready (e.g. it depends on other PRs not yet merged), please mark it as draft until it is ready.

  PRs should meet these guidelines:
  - Each PR should address ONE topic and have an associated issue.
  - No hard-coded paths or personal directories.
  - No temporary or backup files should be committed (including logs).
  - Any code that you disabled by being commented out should be removed or reenabled.
-->
# Description
<!-- This description will become the commit message for the PR. -->
<!--
  Solely pointing to an issue is not an adequate description!

  Please use this format for your description:

  Describe your changes. Focus on the *what* and *why*. The *how* will be evident from the changes. In particular, be sure to note any interface changes, such as command line syntax, that will need to be communicated to users.

  At the end of your description, please be sure to add the issue this PR solves using the word "Resolves". If there are any issues that are related but not yet resolved (including in other repos), you may use "Refs".

  Resolves #1234
  Refs #4321
  Refs NOAA-EMC/repo#5678
-->

This PR adds full capability to produce 24-hr ocean and ice products. In particular, this PR fixes a bug where the CICE history files are missing when `cycle` is not `t00z` and when `FHOUT_ICE` is `24`. This fix has been made possible since the changes from [UFSWM PR #2437](https://github.com/ufs-community/ufs-weather-model/pull/2437) have been merged and the GW PR #3145 updated the UFSWM hash to include these UFSWM updates for global-workflow. 

Resolves #2674 

<!-- For more on writing good commit messages, see https://cbea.ms/git-commit/ -->

# Type of change
- [x] Bug fix (fixes something broken)
- [ ] New feature (adds functionality)
- [ ] Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
<!-- Choose YES or NO from each of the following and delete the other -->
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO
- Does this change require an update to any of the following submodules? YES
  - [ ] EMC verif-global <!-- NOAA-EMC/EMC_verif-global#1234 -->
  - [ ] GDAS <!-- NOAA-EMC/GDASApp#1234 -->
  - [x] GFS-utils <!-- NOAA-EMC/gfs-utils#1234 --> [NOAA-EMC/gfs-utils#107](https://github.com/NOAA-EMC/gfs-utils/issues/107)
  - [ ] GSI <!-- NOAA-EMC/GSI#1234 -->
  - [ ] GSI-monitor <!-- NOAA-EMC/GSI-Monitor#1234 -->
  - [ ] GSI-utils <!-- NOAA-EMC/GSI-Utils#1234 -->
  - [ ] UFS-utils <!-- ufs-community/UFS_UTILS#1234 -->
  - [ ] UFS-weather-model <!-- ufs-community/ufs-weather-model#1234 -->
  - [ ] wxflow <!-- NOAA-EMC/wxflow#1234 -->

# How has this been tested?
<!-- Please list any test you conducted, including the machine.

Example:
- Clone and build on WCOSS
- Cycled test on Orion
- Forecast-only on Hera
-->

- Clone, built and ran on WCOSS2 with the GEFS CI test

# Checklist
- [ ] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have documented my code, including function, input, and output descriptions
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] This change is covered by an existing CI test or a new one has been added
- [ ] Any new scripts have been added to the .github/CODEOWNERS file with owners
- [ ] I have made corresponding changes to the system documentation if necessary

EricSinsky-NOAA committed:
Add support for 24hr-averaged ocn/ice products

EricSinsky-NOAA committed:
Address shellnorms error

EricSinsky-NOAA committed:
Merge remote-tracking branch 'origin/develop' into gw_issue2674

EricSinsky-NOAA committed:
More shellnorm fixes

EricSinsky-NOAA committed:
Make CICE namelist changes specific to GEFS

EricSinsky-NOAA commented on ush/parsing_namelists_CICE.sh:
```diff
@@ -70,7 +70,12 @@ local CICE_RESTART_FORMAT="pnetcdf2"
 local CICE_DUMPFREQ_N=10000  # Set this to a really large value, as cice, mom6 and cmeps restart interval is controlled by ufs.configure
 local CICE_DIAGFREQ=$(( 86400 / DT_CICE ))  # frequency of diagnostic output in timesteps, recommended for 1x per day
-local CICE_HISTFREQ_N="0, 0, ${FHOUT_ICE}, 0, 1"
+if [[ "${RUN}" == "gefs" ]]; then
+  local CICE_DIAGFREQ1=$((( FHOUT_ICE * 3600 )/ DT_CICE ))
+  local CICE_HISTFREQ_N="0, 0, 0, ${CICE_DIAGFREQ1}, 1"
+else
+  local CICE_HISTFREQ_N="0, 0, ${FHOUT_ICE}, 0, 1"
+fi
```
@NeilBarton-NOAA Will SFS have 24-hr ice products as well? If yes, I can add SFS here as well.

NeilBarton-NOAA replied:
@EricSinsky-NOAA Yes, SFS is defaulting to 24 means. We may have additional monthly output, but this is fine for now

EricSinsky-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into gw_issue2674

EricSinsky-NOAA committed:
Add 24 hr ice products to ice_prod task

EricSinsky-NOAA committed:
Add support for an ocean_gefs.csv file

EricSinsky-NOAA committed:
Update gfs_utils hash

The gfs_utils hash has been updated so that the gefs-specfic ocean and ice csv files are added.

EricSinsky-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into gw_issue2674

EricSinsky-NOAA commented:
The GEFS 24-hr ocean and ice products were successfully created when the 12z C48 GEFS offline CI test was performed.  Marking as ready for review.

EricSinsky-NOAA committed:
Add description of new parsing CICE nml variable

NickSzapiro-NOAA commented:
It's an edge case but are timesteps always chosen to be evenly divisible into 24 hours?

fwiw, I think there's an alternate solution to this in CICE via histfreq_base, if needed

aerorahul commented on sorc/link_workflow.sh:
```diff
@@ -154,7 +154,7 @@ for dir in gfs gefs sfs
   ${LINK_OR_COPY} "${HOMEgfs}/sorc/upp.fd/parm/${dir}" .
 done
-for file in ice.csv ocean.csv ocnicepost.nml.jinja2; do
+for file in ice.csv ice_gefs.csv ocean.csv ocean_gefs.csv ocnicepost.nml.jinja2; do
```
Should update to link `ocean_gfs.csv` and `ice_gfs.csv`

EricSinsky-NOAA replied:
@aerorahul This suggestion and your other suggestions have been committed. After some offline testing on Friday, I believe this PR is ready to move forward.

aerorahul commented on .gitignore:
```diff
@@ -74,7 +74,9 @@ parm/post/gfs
 parm/post/gefs
 parm/post/sfs
 parm/post/ocean.csv
+parm/post/ocean_gefs.csv
```
I think we created `ocean_gfs.csv` and `ice_gfs.csv` as well in `gfs_utils`

aerorahul commented on parm/post/oceanice_products_gefs.yaml:
```diff
@@ -10,7 +10,7 @@ ocnicepost:
       - ["{{ EXECgfs }}/ocnicepost.x", "{{ DATA }}/"]
       - ["{{ PARMgfs }}/post/ocnicepost.nml.jinja2", "{{ DATA }}/"]
-      - ["{{ PARMgfs }}/post/{{ component }}.csv", "{{ DATA }}/"]
+      - ["{{ PARMgfs }}/post/{{ component }}_gefs.csv", "{{ DATA }}/{{ component }}.csv"]
```
should the `oceanice_products_gfs.yaml` be updated to be consistent with the changes?

aerorahul reviewed (commented):
some feedback.

EricSinsky-NOAA committed:
Link ocean_gfs.csv and ice_gfs.csv

EricSinsky-NOAA committed:
Rename oceanice_products yaml file

The oceanice_products.yaml file has been renamed to oceanice_products_gfs.yaml to mantain consistency with the oceanice_products_gefs.yaml.

EricSinsky-NOAA committed:
Copy ocean_gfs csv file as ocean.csv to DATA dir

In the oceanice_products_gfs.yaml file, copy ocean_gfs and ice_gfs csv files as ocean and ice csv files into the DATA directory.

EricSinsky-NOAA committed:
Modify oceanice_products config file

The oceanice_products config file has been modified so that it points to the _gfs yaml file.

EricSinsky-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into gw_issue2674

EricSinsky-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into gw_issue2674

EricSinsky-NOAA committed:
Merge remote-tracking branch 'EMC/develop' into gw_issue2674

aerorahul committed:
Merge branch 'develop' into gw_issue2674

DavidHuber-NOAA committed:
Merge branch 'develop' into gw_issue2674

DavidHuber-NOAA committed:
Merge branch 'develop' into gw_issue2674

DavidHuber-NOAA commented:
Relaunching CI on Gaea C6 now that the GSI fix issue has been resolved.

EricSinsky-NOAA commented:
It seems the CI failure of `C96C48mx500_S2SW_cyc_gfs` is related to issue #3627.

DavidHuber-NOAA commented:
@EricSinsky-NOAA Yes, I believe your are correct.  Since this is the only job that failed, I believe this PR is ready to merge.  We can address 3627 in another PR.
